src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -979,7 +979,7 @@ private IOperation CreateBoundConversionOperation(BoundConversion boundConversio
 
                 if (boundConversion.Type is FunctionPointerTypeSymbol)
                 {
-                    Debug.Assert(boundConversion.Conversion.MethodSymbol is object);
+                    Debug.Assert(boundConversion.SymbolOpt is object);
```

Comment:
why remove the previous assertion?

Reply:
They're the same symbol, `SymbolOpt` just returns `Conversion.MethodSymbol` and this makes the flow analysis work better with the rest of the method.